src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetectionExecutor.swift

```diff
@@ -112,9 +112,26 @@ final class ExposureDetectionExecutor: ExposureDetectionDelegate {
 		let rootDir = fileManager.temporaryDirectory.appendingPathComponent(UUID().uuidString, isDirectory: true)
 		do {
 			try fileManager.createDirectory(at: rootDir, withIntermediateDirectories: true, attributes: nil)
-			let packages = downloadedPackagesStore.allPackages(for: .formattedToday(), onlyHours: store.hourlyFetchingEnabled)
-			let writer = AppleFilesWriter(rootDir: rootDir, keyPackages: packages)
-			return writer.writeAllPackages()
+			let writer = AppleFilesWriter(rootDir: rootDir)
+
+			if store.hourlyFetchingEnabled {
```

Comment:
I think we should get rid of all these states… Testing single functions becomes a real mess. How about this for the function signature:

```swift
func exposureDetectionWriteDownloadedPackages(_ detection: ExposureDetection, useHourlyPackages: Bool = false) -> WrittenPackages? {
```

Reply:
To change this I would need to introduce Store to ExposureDetection. Not sure if this is good. And also not sure if this change should be in the scope of this story. What do you mean by a mess? I think you can inject a fake store to ExposureDetectionExecutor.

Reply:
It's becoming messy because we 
  1. have to initialize some fake store and 
  2. take care/know of potential implementations within a function

I just wanted to raise awareness that we have generally a large amount of properties/states in our code and should question some of them. Apart from that I'm fine with the PR.

Reply:
Ok, got you point! Thanks!